src/router/router.js

```diff
@@ -108,8 +111,20 @@ export const navigate = async function () {
     const previousRoute = currentRoute
     const hash = getHash()
     const route = matchHash(hash, this.parent[symbols.routes])
-
+    let beforeHookOutput
     if (route) {
+      if (route.hooks) {
+        if (route.hooks.before) {
+          const args = route.params ? route.params : {}
+          beforeHookOutput = await route.hooks.before(args)
```

Comment:
I would not declare a constant for this. Instead we can probably just inline `route.params || {}` inside `before`-function.

In fact, does it actually happen that params is not an empty object? Since on L79 params are reduced from an empty object literal

Reply:
Will remove constant declaration.
If a route path does not contain any dynamic route paths` route.path.indexOf(':')== -1`, then the params key is not available for such routes.

Reply:
Yeah, you're right. Maybe we should fix this in the `matchHash` function instead then. That seems like a cleaner solution.